### PR TITLE
Show only lois in the selected job in loi editor.

### LINKED ISSUE #1713: [LOI Editor] Map shows LOIs for all jobs
@nwkotto @rfontanarosa @DaoyuT  FYI

Map should only show predefined LOIs in the select job

### PR DESCRIPTION
**Pull request by DaoyuT**
Fix #1713

### CODE REVIEW

File: web/src/app/pages/main-page-container/main-page/map/map.component.ts [hunk 1]
```diff
@@ -124,17 +126,29 @@ export class MapComponent implements AfterViewInit, OnDestroy {
   }
 
   ngAfterViewInit() {
+    this.initMap();
+  }
+
+  ngOnChanges() {
+    this.initMap();
```
Comment (gino-m) [inline web/src/app/pages/main-page-container/main-page/map/map.component.ts:133]: If we call `initMap()` multiple times, we'll accumulate one subscription for each change. Instead, there are two approaches you could take:

1. Do what you're doing here, but also dispose of the previous subscription before creating a new one, or
2. Create an Observable, selectedJobId$, which you combine with the other streams in `initMap()`.

Wdyt?

File: web/src/app/pages/main-page-container/main-page/map/map.component.ts [hunk 2]
```diff
@@ -124,17 +126,29 @@ export class MapComponent implements AfterViewInit, OnDestroy {
...
         this.navigationService.getLocationOfInterestId$(),
       ]).subscribe(([survey, lois, locationOfInterestId]) => {
         const loisMap = ImmutableMap(
-          (this.showPredefinedLoisOnly
-            ? lois.filter(loi => loi.predefined !== false)
-            : lois
-          ).map(loi => [loi.id, loi])
+          lois
+            .filter(loi =>
+              this.showPredefinedLoisOnly ? loi.predefined !== false : loi
```
Comment (gino-m) [inline web/src/app/pages/main-page-container/main-page/map/map.component.ts:146]: Can `loi` ever be falsey? If so, why not just return `loi => loi.predefined !== false` as before?

File: web/src/app/pages/main-page-container/main-page/map/map.component.ts [hunk 3]
```diff
@@ -124,17 +126,29 @@ export class MapComponent implements AfterViewInit, OnDestroy {
...
+            )
+            .filter(loi =>
+              this.selectedJobId !== '' ? loi.jobId === this.selectedJobId : loi
```
Comment (gino-m) [inline web/src/app/pages/main-page-container/main-page/map/map.component.ts:149]: Same question here. Can this be rewritten without evaluating the truthiness of `loi`?

**Review by gino-m — CHANGES_REQUESTED** [review]

File: web/src/app/pages/main-page-container/main-page/map/map.component.ts (hunk 2 above)
Comment (DaoyuT) [reply web/src/app/pages/main-page-container/main-page/map/map.component.ts:146]: RIght, I don't know why I wrote it that way, changed.

File: web/src/app/pages/main-page-container/main-page/map/map.component.ts (hunk 3 above)
Comment (DaoyuT) [reply web/src/app/pages/main-page-container/main-page/map/map.component.ts:149]: Done.

**Review by gino-m — DISMISSED** [review]
LGTM! 🚀

File: web/src/app/pages/main-page-container/main-page/map/map.component.ts (hunk 1 above)
Comment (DaoyuT) [reply web/src/app/pages/main-page-container/main-page/map/map.component.ts]: Agreed, took option 2.

**Review by gino-m — APPROVED** [review]

**Pull request merged**